### PR TITLE
Simple KNN search unit test over natural numbers

### PR DESCRIPTION
**Pull request by turambar**
Added a simple KNN search unit test over natural numbers. Currently fails, at least on my machine.

## What changes were proposed in this pull request?

See above.

## How was this patch tested?

It's a unit test. ;)

## Quick checklist

The following checklist helps ensure your PR is complete:

- [x] Reviewed the [Contributing Guidelines](https://github.com/deeplearning4j/deeplearning4j/blob/master/CONTRIBUTING.md) and followed the steps within.
- [x] Created tests for any significant new code additions.
- [ ] Relevant tests for your changes are passing.
- [ ] Ran mvn formatter:format (see [formatter instructions](http://code.revelc.net/formatter-maven-plugin/examples.html#Setting_Source_Files) for targeting your specific files).

### CODE REVIEW

**Comment by turambar** [comment]
Would appreciate a sanity check on this @agibsonccc @raver119 @emsixteeen. Given your shared confidence this is fixed, I wonder whether it's me or you that is delusional!

File: deeplearning4j-nearestneighbors-parent/nearestneighbor-core/src/test/java/org/deeplearning4j/clustering/vptree/VpTreeNodeTest.java [hunk 1]
```diff
@@ -148,4 +145,44 @@ public void vpTreeTest() {
...
+        final int queryPoint = 12;
+        final int k = 5;
+
+        INDArray points = Nd4j.arange(firstPoint, lastPoint).transpose();
+        INDArray query = Nd4j.zeros(1);
+        query.putScalar(0, queryPoint);
+
+        INDArray trueResults = Nd4j.zeros(k);
+        for (int i = 0; i < k; i++) {
+            int pt = queryPoint - k / 2 + i;
```
Comment (huitseeker) [inline deeplearning4j-nearestneighbors-parent/nearestneighbor-core/src/test/java/org/deeplearning4j/clustering/vptree/VpTreeNodeTest.java]: you may want to explicitly say `10 + i` for clarity

File: deeplearning4j-nearestneighbors-parent/nearestneighbor-core/src/test/java/org/deeplearning4j/clustering/vptree/VpTreeNodeTest.java [hunk 2]
```diff
@@ -148,4 +145,44 @@ public void vpTreeTest() {
...
+        VPTree tree = new VPTree(points);
+        List<DataPoint> results = new ArrayList<>();
+        List<Double> distances = new ArrayList<>();
+        tree.search(query, k, results, distances);
+        INDArray sortedResults = Nd4j.zeros(k);
+        int i = 0;
+        for (DataPoint p : results)
+            sortedResults.put(new int[] {0, i++}, p.getPoint());
+        Nd4j.sort(sortedResults, 1, true);
+        if (!trueResults.equalsWithEps(sortedResults, 1E-12))
```
Comment (huitseeker) [inline deeplearning4j-nearestneighbors-parent/nearestneighbor-core/src/test/java/org/deeplearning4j/clustering/vptree/VpTreeNodeTest.java]: You should use junit's `assertTrue` and assertEquals`, see:
http://junit.org/junit4/javadoc/4.8/org/junit/Assert.html

They take a String argument for displaying a message, in which you can print the received values (as opposed to expected)

File: deeplearning4j-nearestneighbors-parent/nearestneighbor-core/src/test/java/org/deeplearning4j/clustering/vptree/VpTreeNodeTest.java [hunk 3]
```diff
@@ -148,4 +145,44 @@ public void vpTreeTest() {
...
+
+        VPTreeFillSearch fillSearch = new VPTreeFillSearch(tree, k, query);
+        fillSearch.search();
+        results = fillSearch.getResults();
+        sortedResults = Nd4j.zeros(k);
+        i = 0;
+        for (DataPoint p : results)
+            sortedResults.put(new int[] {0, i++}, p.getPoint());
+        Nd4j.sort(sortedResults, 1, true);
+        if (!trueResults.equalsWithEps(sortedResults, 1E-12))
```
Comment (huitseeker) [inline deeplearning4j-nearestneighbors-parent/nearestneighbor-core/src/test/java/org/deeplearning4j/clustering/vptree/VpTreeNodeTest.java]: See comment above.

**Pull request merged**